packages/job-manager/src/batch/batch.ts

```diff
@@ -26,6 +28,61 @@ export class Batch {
         @inject(Logger) private readonly logger: Logger,
     ) {}
 
+    public async getPoolTaskProcessingRatio(): Promise<number> {
+        const resultVariableName = 'taskProcessingRatioResult';
+        const poolTaskProcessingRatioFormula = `
+        // The script will calculate the tasks processing ratio returned in $taskProcessingRatio variable as
+        // double value and will return -1 when there are no active tasks or sampling data is insufficient
+
+        // the time interval for data sampling
+        $sampleTimeIntervalMinutes = 15 * TimeInterval_Minute;
+        $samplePercentThreshold = 70;
+
+        // get sample data for pending tasks (the sum of active and running tasks)
+        $pendingTasksSamplePercent = $PendingTasks.GetSamplePercent($sampleTimeIntervalMinutes);
+        $pendingTasksVector = $PendingTasks.GetSample(TimeInterval_Minute, $sampleTimeIntervalMinutes);
```

Comment:
the documentation says, GetSample may fail. does this mean, it may throw? if so, do we need to handle that? 

(under https://docs.microsoft.com/en-us/azure/batch/batch-automatic-scaling#getsampledata)

Because the GetSample method fails if the percentage of samples returned is less than the samplePercent specified, you can use the GetSamplePercent method to check first. Then you can perform an alternate action if insufficient samples are present, without halting the automatic scaling evaluation.

Reply:
The formula evaluation will not throw. The documentation you referenced explains case where samplePercent parameter is specified. The current code does not have that parameter.

---
In reply to: [305942247](https://github.com/microsoft/accessibility-insights-service/pull/221#discussion_r305942247) [](ancestors = 305942247)